### PR TITLE
Fix exception when unhandled exception occurs after shutdown

### PR DESCRIPTION
**Pull request by MatkovIvan**

### CODE REVIEW

**Comment by achocron** [comment]
Sorry for the constant closing/reopening, CI is not starting builds correctly.

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs [hunk 1]
```diff
@@ -292,6 +294,18 @@ private void SetInstanceCustomProperties(CustomProperties customProperties)
             _channel.EnqueueAsync(customPropertiesLog);
         }
 
+        private void OnUnhandledExceptionOccurred(object sender, UnhandledExceptionOccurredEventArgs args)
+        {
+            lock (AppCenterLock)
+            {
+                if (_channelGroup != null)
+                {
+                    _channelGroup.ShutdownAsync();
```
Comment (achocron) [inline SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs]: Can you add a debug log that says "Shutting down App Center due to unhandled exception"?

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Channel/ChannelGroup.cs [hunk 2]
```diff
@@ -99,6 +100,15 @@ public void SetEnabled(bool enabled)
 
         public async Task ShutdownAsync()
         {
+            lock (_channelGroupLock)
```
Comment (achocron) [inline SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Channel/ChannelGroup.cs]: Just out of curiosity, why is this before the call to `ThrowIfDisposed()`?

Comment (MatkovIvan) [reply SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Channel/ChannelGroup.cs]: Doesn't matter I guess. Will change, not big deal

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs (hunk 1 above)
Comment (MatkovIvan) [reply SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs]: "Shutting down **channel group** due to unhandled exception."

**Pull request merged**